### PR TITLE
tiledb: 2.3.3 -> 2.8.3, spdlog: 1.10.0 -> 1.11.0, spdlog_0: remove

### PR DESCRIPTION
**Pull request by panicgh**
###### Description of changes

- update tiledb to a version compatible with spdlog, unpin from spdlog_0.
  More recent versions of tiledb exist, but 2.8.3 is the last where the dependencies are easy to handle. I didn't manage to build tiledb >= 2.9.0 because of cmake errors and mixed internal and external dependencies. So I chose the "quick" route and bumped the version of tiledb just a bit and not to the latest possible.
- remove spdlog_0 because it is not used anymore
- update spdlog to the latest release
- cleanup micromamba from a workaround where it used fmt_9 and spdlog used fmt_8. Align both to fmt_9.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux (`pkgsCross.aarch64-multiplatform.spdlog`)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [X] x86_64-windows (`pkgsCross.mingwW64.spdlog`)
  - [X] i686-windos (`pkgsCross.mingw32.spdlog`)
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KiruyaMomochi** [comment]
Result of `nixpkgs-review pr 222632` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>qlandkartegt</li>
  </ul>
</details>
<details>
  <summary>59 packages failed to build:</summary>
  <ul>
    <li>irods</li>
    <li>irods-icommands</li>
    <li>lizardfs</li>
    <li>mapcache</li>
    <li>openroad</li>
    <li>perl534Packages.Tirex</li>
    <li>perl534Packages.Tirex.devdoc</li>
    <li>perl536Packages.Tirex</li>
    <li>perl536Packages.Tirex.devdoc</li>
    <li>poac</li>
    <li>pynitrokey</li>
    <li>pynitrokey.dist</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.cartopy.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-aer.dist</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-finance.dist</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ibmq-provider.dist</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-ignis.dist</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-nature.dist</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-optimization.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.tiledb</li>
    <li>python310Packages.tiledb.dist</li>
    <li>worldengine-cli (python310Packages.worldengine)</li>
    <li>worldengine-cli.dist (python310Packages.worldengine.dist)</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.pc-ble-driver-py</li>
    <li>python311Packages.pc-ble-driver-py.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.qiskit-aer</li>
    <li>python311Packages.qiskit-aer.dist</li>
    <li>python311Packages.qiskit-finance</li>
    <li>python311Packages.qiskit-finance.dist</li>
    <li>python311Packages.qiskit-ibmq-provider</li>
    <li>python311Packages.qiskit-ibmq-provider.dist</li>
    <li>python311Packages.qiskit-ignis</li>
    <li>python311Packages.qiskit-ignis.dist</li>
    <li>python311Packages.qiskit-nature</li>
    <li>python311Packages.qiskit-nature.dist</li>
    <li>python311Packages.qiskit-optimization</li>
    <li>python311Packages.qiskit-optimization.dist</li>
    <li>python311Packages.tiledb</li>
    <li>python311Packages.tiledb.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>t-rex</li>
  </ul>
</details>
<details>
  <summary>133 packages built:</summary>
  <ul>
    <li>aitrack</li>
    <li>ananicy-cpp</li>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>bear</li>
    <li>cherrytree</li>
    <li>cloudcompare</li>
    <li>coeurl</li>
    <li>cryfs</li>
    <li>entwine</li>
    <li>far2l</li>
    <li>gdal (python310Packages.gdal)</li>
    <li>gerbera</li>
    <li>gmt</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradioMinimal (gnuradio3_9Minimal)</li>
    <li>gnuradioPackages.gnuradio</li>
    <li>gplates</li>
    <li>grass</li>
    <li>hal-hardware-analyzer</li>
    <li>haskellPackages.hgdal</li>
    <li>haskellPackages.hgdal.doc</li>
    <li>iptsd</li>
    <li>kodi (kodiPackages.kodi)</li>
    <li>kodi-gbm</li>
    <li>kodi-wayland</li>
    <li>kodiPackages.a4ksubtitles</li>
    <li>kodiPackages.archive_tool</li>
    <li>kodiPackages.iagl</li>
    <li>kodiPackages.inputstream-adaptive</li>
    <li>kodiPackages.inputstream-ffmpegdirect</li>
    <li>kodiPackages.inputstream-rtmp</li>
    <li>kodiPackages.invidious</li>
    <li>kodiPackages.joystick</li>
    <li>kodiPackages.kodi-platform</li>
    <li>kodiPackages.libretro</li>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>kodiPackages.netflix</li>
    <li>kodiPackages.pvr-hdhomerun</li>
    <li>kodiPackages.pvr-hts</li>
    <li>kodiPackages.pvr-iptvsimple</li>
    <li>kodiPackages.steam-controller</li>
    <li>kodiPackages.vfs-libarchive</li>
    <li>kodiPackages.vfs-sftp</li>
    <li>kodiPackages.visualization-waveform</li>
    <li>libLAS</li>
    <li>libsForQt5.plasma-vault (plasma5Packages.plasma-vault)</li>
    <li>lokinet</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>micromamba</li>
    <li>mtxclient</li>
    <li>mysql-workbench</li>
    <li>nheko</li>
    <li>nrfutil</li>
    <li>nrfutil.dist</li>
    <li>octavePackages.mapping</li>
    <li>openorienteering-mapper</li>
    <li>paraview</li>
    <li>pc-ble-driver</li>
    <li>pdal</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql11Packages.postgis.doc</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql12Packages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresqlPackages.postgis (postgresql14Packages.postgis)</li>
    <li>postgresqlPackages.postgis.doc (postgresql14Packages.postgis.doc)</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.fiona.dist</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.folium.dist</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.geopandas.dist</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.osmnx.dist</li>
    <li>python310Packages.pc-ble-driver-py</li>
    <li>python310Packages.pc-ble-driver-py.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.pygmt</li>
    <li>python310Packages.pygmt.dist</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.python-mapnik.dist</li>
    <li>python310Packages.rasterio</li>
    <li>python310Packages.rasterio.dist</li>
    <li>python310Packages.vowpalwabbit</li>
    <li>python310Packages.vowpalwabbit.dist</li>
    <li>python310Packages.wktutils</li>
    <li>python310Packages.wktutils.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.vowpalwabbit</li>
    <li>python311Packages.vowpalwabbit.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>saga</li>
    <li>spdlog</li>
    <li>spdlog.dev</li>
    <li>spdlog.doc</li>
    <li>sumo</li>
    <li>tiledb</li>
    <li>udig</li>
    <li>vowpal-wabbit</li>
    <li>wasmedge</li>
    <li>waybar</li>
    <li>wsysmon</li>
  </ul>
</details>

**Comment by panicgh** [comment]
Comparison with a recent Hydra nixkgs:trunk [evaluation](https://hydra.nixos.org/eval/1792558):
<details>

> 59 packages failed to build:
>    * irods
>    * irods-icommands

fail

>    * lizardfs

not built on Hydra

>    * mapcache

**succeeds** only on Darwin, fails on Linux

>    * openroad

**succeeds**

>    * perl534Packages.Tirex
>    * perl534Packages.Tirex.devdoc
>    * perl536Packages.Tirex
>    * perl536Packages.Tirex.devdoc

fail

>    * poac

**succeeds**

>    * pynitrokey
>    * pynitrokey.dist

not built

>    * python310Packages.cartopy
>    * python310Packages.cartopy.dist

fail

>    * python310Packages.qiskit
>    * python310Packages.qiskit-aer
>    * python310Packages.qiskit-aer.dist
>    * python310Packages.qiskit-finance
>    * python310Packages.qiskit-finance.dist
>    * python310Packages.qiskit-ibmq-provider
>    * python310Packages.qiskit-ibmq-provider.dist
>    * python310Packages.qiskit-ignis
>    * python310Packages.qiskit-ignis.dist
>    * python310Packages.qiskit-nature
>    * python310Packages.qiskit-nature.dist
>    * python310Packages.qiskit-optimization
>    * python310Packages.qiskit-optimization.dist
>    * python310Packages.qiskit.dist
>    * python310Packages.rlax
>    * python310Packages.rlax.dist
>    * python310Packages.tiledb
>    * python310Packages.tiledb.dist

not built

>    * worldengine-cli (python310Packages.worldengine)
>    * worldengine-cli.dist (python310Packages.worldengine.dist)

fail due to failing dependency

>    * python311Packages.cartopy
>    * python311Packages.cartopy.dist

fail

>    * python311Packages.pc-ble-driver-py
>    * python311Packages.pc-ble-driver-py.dist

not built

>    * python311Packages.plotnine
>    * python311Packages.plotnine.dist
>    * python311Packages.python-mapnik
>    * python311Packages.python-mapnik.dist

fail for python311

>    * python311Packages.qiskit-aer
>    * python311Packages.qiskit-aer.dist
>    * python311Packages.qiskit-finance
>    * python311Packages.qiskit-finance.dist
>    * python311Packages.qiskit-ibmq-provider
>    * python311Packages.qiskit-ibmq-provider.dist
>    * python311Packages.qiskit-ignis
>    * python311Packages.qiskit-ignis.dist
>    * python311Packages.qiskit-nature
>    * python311Packages.qiskit-nature.dist
>    * python311Packages.qiskit-optimization
>    * python311Packages.qiskit-optimization.dist
>    * python311Packages.tiledb
>    * python311Packages.tiledb.dist

not built

>    * python311Packages.worldengine
>    * python311Packages.worldengine.dist


fail due to failing dependency

>    * t-rex

fail
</details>

Those pkgs which succeed on Hydra but fail with the new spdlog are:
* mapcache (succeeds only on Darwin)
* openroad
* poac

**Comment by mausch** [comment]
Tested micromamba at c0d17fc749a598445b41101cf471c7cbb4899287 , seems to work fine :+1:

**Review by mausch — APPROVED** [review]

**Comment by panicgh** [comment]
I created #224312 to bump openroad. When it is merged, it will also build with this PR, i.e. there should be no issues with the spdlog update any more.

**Comment by panicgh** [comment]
Openroad has since been bumped and now works with the new spdlog/fmt. Then I rebased this PR's branch and selectively rebuilt the 4 packages:

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>micromamba</li>
    <li>openroad</li>
    <li>poac</li>
    <li>tiledb</li>
  </ul>
</details>

File: pkgs/development/libraries/spdlog/default.nix [hunk 1]
```diff
@@ -2,67 +2,50 @@
...
+    repo  = "spdlog";
+    rev   = "v${version}";
+    hash  = "sha256-c6s27lQCXKx6S1FhZ/LiKh14GnXMhZtD1doltU4Avws=";
   };
 
-  spdlog_0 = generic {
-    version = "0.17.0";
-    sha256 = "112kfh4fbpm5cvrmgbgz4d8s802db91mhyjpg7cwhlywffnzkwr9";
+  # in master post 1.10.0, see https://github.com/gabime/spdlog/issues/2380
+  patches = lib.optional (lib.versionAtLeast version "1.4.1") (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/spdlog/default.nix]:
```suggestion
 patches = [
   (fetchpatch {
```
thats now always true

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Pull request merged**